lib/timetable-form/Form.tsx

```diff
@@ -1,10 +1,61 @@
+import { useState } from "react";
 import styles from "./styles.module.css";
 
+const stageNames = [
+  "Timetable published",
+  "Draft plan for public consultation published",
+  "Public consultation start",
+  "Public consultation end",
+  "Submit plan for examination",
+  "Examination hearing start",
+  "Planning inspectorate examination start",
+  "Planning inspectorate examination end",
+  "Planning inspectorate found sound",
+  "Examination hearing end",
+  "Inspector report published",
+  "Plan adopted",
+] as const;
+
+type StageName = (typeof stageNames)[number];
+
+const defaultDate = "";
+
+type Stages = {
+  [key in StageName]: string;
+};
+
+const initialStages = Object.fromEntries(
+  stageNames.map((stage) => [stage, defaultDate])
+) as Stages;
```

Comment:
I wonder if we're better off keep this as an array so that we're 100% sure we're maintaining the order?

Reply:
Do you mean having the "stages" state as an array of [stageName, stageDate]?

Reply:
That's what I meant, yeah. That said, I think we're probably safe relying on the object key order here, just something I wanted to raise because the order isn't guaranteed like it is for an array.

Reply:
if we want to make sure that display stages in the same order as in the array we can change the mapping inside the JSX to be done on the stageNames array rather than on Object.entries(stages)

Reply:
OK, yeah, I like that approach, it also keeps updating simple. One quick thought on the stages is that currently we only have the event name but not the id/code that we'll need for saving in CSV rows?

Reply:
Agreed. That will be necessary for the download CSV feature so we can probably tackle it as part of that [ticket](https://trello.com/c/60P86fmr/10-user-story-download-form-data)

Reply:
It might make sense to tackle it now because it change the structure of the component a bit but I'm happy to tackle later 👍